include/boost/graph/link_cut_trees.hpp

```diff
@@ -0,0 +1,280 @@
+//
+//=======================================================================
+// Copyright 2019
+// Author: Yi Ji
+//
+// Distributed under the Boost Software License, Version 1.0. (See
+// accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+//=======================================================================
+//
+#ifndef BOOST_LINK_CUT_TREES_HPP
+#define BOOST_LINK_CUT_TREES_HPP
+
+#include <boost/graph/properties.hpp>
+#include <boost/unordered_map.hpp>
+
+namespace boost
+{
+    template<class ElementParentMap, class ElementChildMap = ElementParentMap>
+    class link_cut_trees
+    {
+    public:
+        inline link_cut_trees(ElementParentMap p, ElementChildMap l, ElementChildMap r) : parent(p), left(l), right(r) {}
+        
+        inline link_cut_trees(const link_cut_trees<ElementParentMap, ElementChildMap> &c)
```

Comment:
This is an unimportant topic but you might like to know that you can just use the class name (e.g. `link_cut_trees`) inside the class without the template parameters and it will automatically use the current values of them. It can help make code a little more readable.

Reply:
Good to know! Fixed it.